File: backend/app/schemas/voice.py
from typing import Optional, List
from pydantic import BaseModel, ConfigDict

class ParsedCommandSchema(BaseModel):
    model_config = ConfigDict(use_enum_values=True)

    intent: str
    product_text: Optional[str] = None
    product_id: Optional[str] = None
    product_name: Optional[str] = None
    quantity: Optional[float] = None
    unit: Optional[str] = None
    price_total: Optional[float] = None
    confidence: float = 0.0

class VoiceCommandResponse(BaseModel):
    interaction_id: str
    status: str
    transcript: str
    language: str
    command: ParsedCommandSchema
    confirmation_text: Optional[str] = None

class VoiceCommitRequest(BaseModel):
    confirmed: bool = True
    product_id: Optional[str] = None
    operation: Optional[str] = None
    quantity: Optional[float] = None
    unit: Optional[str] = None

class VoiceCommitResponse(BaseModel):
    status: str = "committed"
    transaction_id: Optional[str] = None
    product_name: Optional[str] = None
    new_balance: Optional[float] = None
    message: str

class CandidateSchema(BaseModel):
    id: str
    name: str

File: frontend/src/types/voice.ts
export type VoiceState = 
  | 'idle' 
  | 'listening' 
  | 'recording' 
  | 'uploading' 
  | 'transcribing' 
  | 'understanding' 
  | 'needs_confirmation' 
  | 'committing' 
  | 'completed' 
  | 'error';

export interface ParsedCommand {
  action: 'add' | 'remove' | 'query' | 'unknown';
  product: string;
  product_id?: string;
  product_name?: string;
  quantity?: number;
  unit?: string;
  price?: number;
  confidence: number;
  original_text: string;
}

export interface VoiceCommand {
  id: string;
  transcript: string;
  language: string;
  parsed_intent: ParsedCommand;
  status: 'pending' | 'confirmed' | 'cancelled' | 'failed';
  created_at: string;
}

export interface BackendVoicePreview {
  status: string;
  interaction_id?: string;
  transcript?: string;
  intent?: string;
  message?: string;
  confirmation_text?: string;
  answer?: unknown;
  product_name?: string;
  quantity?: number;
  unit?: string;
  candidates?: Array<{ id: string; name: string }>;
  command?: {
    intent: string;
    product_text?: string;
    product_id?: string;
    product_name?: string;
    quantity?: number;
    unit?: string;
    price_total?: number | null;
    confidence?: number;
  };
  preview?: unknown;
}
